src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs

```diff
@@ -2456,5 +2457,49 @@ public class C
                 Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(5, 49)
                 );
         }
+
+        [Fact]
+        [WorkItem(58073, "https://github.com/dotnet/roslyn/issues/58073")]
+        public void DiagnosticAdditionalLocations_SquiggleConstructor()
+        {
+            var source =
+@"
+public class C
+{
+    public C() { }
+
+    public string S { get; }
+}";
+            var comp = CreateCompilation(source, options: WithNullableEnable());
+            comp.VerifyDiagnostics(
+                // (4,12): warning CS8618: Non-nullable property 'S' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.
+                //     public C() { }
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "C").WithArguments("property", "S").WithLocation(4, 12));
+
+            var property = comp.GetTypeByMetadataName("C").GetMember("S");
+            var actualAdditionalLocations = comp.GetDiagnostics().Single().AdditionalLocations;
+            Assert.Equal(property.Locations.Single(), actualAdditionalLocations.Single());
+        }
+
+        [Fact]
+        [WorkItem(58073, "https://github.com/dotnet/roslyn/issues/58073")]
+        public void DiagnosticAdditionalLocations_SquiggleProperty()
+        {
+            var source =
+@"
+public class C
+{
+    public string S { get; }
+}";
+            var comp = CreateCompilation(source, options: WithNullableEnable());
+            comp.VerifyDiagnostics(
+                // (4,19): warning CS8618: Non-nullable property 'S' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.
+                //     public string S { get; }
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "S").WithArguments("property", "S").WithLocation(4, 19));
+
+            var property = comp.GetTypeByMetadataName("C").GetMember("S");
+            var actualAdditionalLocations = comp.GetDiagnostics().Single().AdditionalLocations;
+            Assert.Equal(property.Locations.Single(), actualAdditionalLocations.Single());
```

Comment:
@Youssef1313 not sure the additional location makes sense when the diagnostic is issued on the property (i.e. no constructor), but I'm not sure what the conventions are here.

Reply:
I'll let someone from the compiler team weigh on how this should behave. My intention was to allow the use of `AdditionalLocations` in all cases.

Reply:
I think not having to think about whether to use AdditionalLocations or not is the correct move.